internal/tjson/schema.go

```diff
@@ -212,9 +212,38 @@ func (s *Schema) Unmarshal(b []byte) error {
 		return err
 	}
 
+	// Add $k properties that are necessary for documents.
+	s.addDocumentProperties()
+
 	return nil
 }
 
+// addDocumentProperties adds missing $k properties to all the schema's documents (top-level and nested).
+func (s *Schema) addDocumentProperties() {
+	if s.Type != Object && s.Type != "" {
```

Comment:
That doesn't cover arrays of objects, right?

Reply:
This is only for either "objects" or high-level objects (when `Type` might be unset).

Reply:
Arrays might be needed too, yes.